### PR TITLE
add keepSameTeams feature

### LINKED ISSUE #77: Add keep same teams checkbox to play again end game screen
When the game is over, the host can choose to play again. If they do it randomly chooses teams again. The goal of this is to give the option to keep same teams. I imagine a checkbox they can check prior to clicking this button. 

**Requirements:**
- Add this as a param on `createGame` method
- update restAPI documentation with this change
- modify client and server

### PR DESCRIPTION
**Pull request by jacten**
This feature adds a "keep same teams" button to the PlayAgain screen. 

Small feature but a useful one for if you are in an intense battle and want to keep it going!

### CODE REVIEW

File: client/src/views/Party/PlayAgain.jsx [hunk 1]
```diff
@@ -1,29 +1,36 @@
...
 
   const handlePlayAgainClick = async () => {
+    setIsSubmitting(true);
+
     try {
       await api.updateSettings({
         slug: party.slug,
         settings: {
-          teamsCount,
+          ...(!keepSameTeams && { teamsCount }),
```
Comment (jgsheppa) [inline client/src/views/Party/PlayAgain.jsx:22]: Why is this necessary?

**Review by jgsheppa — APPROVED** [review]
Looks good! I put one question in about updating the settings, but other than that everything made sense to me.

File: client/src/views/Party/PlayAgain.jsx (hunk 1 above)
Comment (jacten) [reply client/src/views/Party/PlayAgain.jsx:22]: probably would decent idea for me to add a comment here. 

But this makes so you can only change the teamsCount (i.e. the number of teams) if keepSameTeams is not selected. If you had a different number of teams you couldn't keep the same teams. 

This is a pattern I end up using a lot. Looks weird but is a nice way of conditionally add an entry to an object. 

The API is set-up so it will only update the fields that are passed to backend.
